cstool/cstool.c

```diff
@@ -663,19 +685,27 @@ int main(int argc, char **argv)
 		}
 	}
 
+	size_t mode_len = strlen(mode);
+	const char *plus = strchr(mode, '+');
+	if (plus) {
+		mode_len = plus - mode;
+	}
+
 	for (i = 0; all_archs[i].name; i++) {
-		if (!strcmp(all_archs[i].name, mode)) {
+		size_t len = strlen(all_archs[i].name);
+		if (len == mode_len && !strncmp(all_archs[i].name, mode, mode_len)) {
 			arch = all_archs[i].arch;
-			err = cs_open(all_archs[i].arch, all_archs[i].mode, &handle);
+			md = all_archs[i].mode;
```

Comment:
```suggestion
			mode = all_archs[i].mode;
```
And above `mode_arg`/`mode_arg_len` or something.
No need to continue with the imprecise naming from before :)